database/pgx/pgx.go

```diff
@@ -69,27 +67,26 @@ type Config struct {
 
 type Postgres struct {
 	// Locking and unlocking need to use the same connection
-	conn     *sql.Conn
-	db       *sql.DB
+	conn     *pgx.Conn
 	isLocked atomic.Bool
 
 	// Open and WithInstance need to guarantee that config is never nil
 	config *Config
 }
 
-func WithInstance(instance *sql.DB, config *Config) (database.Driver, error) {
+func WithInstance(instance *pgx.Conn, config *Config) (database.Driver, error) {
```

Comment:
This is a breaking change. Can you convert the `sql.DB` to a `pgx.Conn` internally?